### PR TITLE
COSI | CLI | bucketclass & bucketclaim support

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. Added cosi bucket claim and bucket class commands to noobaa cli.
2. Supported create, list, status and delete for both COSI bucketclaim and bucketclass
3. Refactored bucketclass cli in order to use the same populate functions for cosi bucketclass.
4. Added COSI types 
5. Added COSI bucket crd yaml to be used on COSI bucket claim creation.
6. Updated docs
7. Added cli integration tests

### Testing Instructions:
1. make test-cosi

- [x] Doc added/updated
- [x] Tests added

### CODE REVIEW

File: pkg/cosi/cosi_bucket_claim.go [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+}
+
+// WaitReady waits until the cosi bucket claim status bucket ready changes to true
+func WaitReady(cosiBucketClaim *nbv1.COSIBucketClaim) bool {
+	log := util.Logger()
+	klient := util.KubeClient()
+
+	intervalSec := time.Duration(3)
+
+	err := wait.PollImmediateInfinite(intervalSec*time.Second, func() (bool, error) {
```
Comment (jackyalbo) [inline pkg/cosi/cosi_bucket_claim.go:292]: let's try to limit the time since we don't have a failed status like in obc.
in future PR please add a recording of error events so the user will know the failure reason.

File: pkg/cosi/cosi_bucket_class.go [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+	// placement policy flags
+	cmd.Flags().String("placement", "",
+		"Set first tier placement policy - Mirror | Spread | \"\" (empty defaults to single backing store)")
+	cmd.Flags().StringSlice("backingstores", nil,
+		"Set first tier backing stores (use commas or multiple flags)")
+	cmd.Flags().String("replication-policy", "",
+		"Set the json file name that contains the replication rules")
+	cmd.Flags().String("deletion-policy", "delete",
+		"Set the deletion policy of the buckets delete | retain - the default deletion policy is delete")
```
Comment (jackyalbo) [inline pkg/cosi/cosi_bucket_class.go:65]: maybe rephrase this so user will better understand the meaning of the options and the default.

File: pkg/cosi/cosi_bucket_class.go [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+	deletionPolicy, _ := cmd.Flags().GetString("deletion-policy")
+	if deletionPolicy == "delete" {
+		bucketClass.DeletionPolicy = nbv1.COSIDeletionPolicyDelete
+	} else if deletionPolicy == "retain" {
+		bucketClass.DeletionPolicy = nbv1.COSIDeletionPolicyRetain
+	} else {
+		log.Fatalf(`❌ Invalid deletion policy, valid values are delete or retain %s`, deletionPolicy)
+	}
+
+	bucketClassSpec := &nbv1.BucketClassSpec{}
```
Comment (jackyalbo) [inline pkg/cosi/cosi_bucket_class.go:238]: move next to populate function

File: pkg/cosi/cosi_bucket_class.go [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+		pp := bc.Parameters["placementPolicy"]
+		np := bc.Parameters["namespacePolicy"]
+		quota := bc.Parameters["quota"]
+		rp := bc.Parameters["replicationPolicy"]
+		table.AddRow(
+			bc.Name,
+			pp,
+			np,
+			quota,
+			rp,
```
Comment (jackyalbo) [inline pkg/cosi/cosi_bucket_class.go:430]: also here limit the size for replication policy

File: pkg/bucketclass/bucketclass.go [hunk 5]
```diff
@@ -522,11 +535,13 @@ func RunList(cmd *cobra.Command, args []string) {
 		pp, _ := json.Marshal(bc.Spec.PlacementPolicy)
 		np, _ := json.Marshal(bc.Spec.NamespacePolicy)
 		quota, _ := json.Marshal(bc.Spec.Quota)
+		rp, _ := json.Marshal(bc.Spec.ReplicationPolicy)
```
[the remark this answers is not in the document]
Comment (romayalon) [reply pkg/bucketclass/bucketclass.go]: I decided to remove it from this PR as it doesn't look good even with a limited number of chars, we need to think about how to show it in a better way and it's not related to this PR so we can move it out

File: pkg/cosi/cosi_bucket_claim.go (hunk 1 above)
Comment (romayalon) [reply pkg/cosi/cosi_bucket_claim.go:295]: added retries

File: pkg/cosi/cosi_bucket_class.go (hunk 2 above)
Comment (romayalon) [reply pkg/cosi/cosi_bucket_class.go]: I also changed the default to retain as COSI's default

**Review by jackyalbo — APPROVED** [review]
